tests/solana_tests/base58_encoding.rs

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::borsh_encoding::BorshToken;
+use crate::{account_new, build_solidity};
+use base58::ToBase58;
+
+#[test]
+fn print_addresses() {
+    let mut vm = build_solidity(
+        r#"
+contract Base58 {
+    function print_this(address addr) pure public {
+        print("{}".format(addr));
+    }
+}
+        "#,
+    );
+
+    vm.constructor(&[]);
+
+    for _ in 0..10 {
+        let account = account_new();
+        let _ = vm.function("print_this", &[BorshToken::Address(account)]);
+        let mut base_58 = account.to_base58();
+        while base_58.len() < 44 {
+            base_58.insert(0, '1');
+        }
```

Comment:
Why is the output different?

Reply:
Rust's `to_base58` ignores leading zeros in the byte array, so it won't transform them into ones. On the other hand, Solana addresses can start with leading ones: `11128aXFh5abEooZ2ouNDjPjk2TqDaHjG6JkX74vK4q` is a valid address.

Reply:
That deserves a comment